### ISSUE TITLE
No predictions by the model

### ISSUE THREAD

**Issue by anushkasw** [DianboWork/SPN4RE#26]
I am trying to apply this model to some other datasets. I found that at times the model gives no predictions.
I found that it might be happening due to setting the relation logit output shape to num_classes+1
self.decoder2class = nn.Linear(config.hidden_size, num_classes + 1)

and then not creating triples when relation predicted was equal to num_classes
def generate_strategy(pred_rel, pred_head, pred_tail, num_classes, _Pred_Triple):
    if pred_rel.pred_rel != num_classes:

Can somebody help me understand why this is done this way?

**Comment by DianboWork** [comment]
Different sentences can mentions different number of triples, so we need to involve a NA relation to indicate no
triple. To fully understand that, you need to read the paper (Bipartite Matching Loss Section).
